Fixed wrong marginal function.

thogerig opened the pull request:
The marginal function to mask a MoMo gave wrong results. The reason was that the masked mesh can result in a different amount of PointIds. Since the mask MoMo function depends on the mesh in contrast to the marginal function for the GP, I would propose to rename it to  **mask()**.

thogerig committed:
Fixed marginal function error: After mesh masking the point ids did not correspond anymore.

BernhardEgger commented:
See https://github.com/unibas-gravis/scalismo-faces/pull/95 for previous discussions

Andreas-Forster reviewed (commented):
Thank you for pointing out this bug. However, I also do not really like the new solution. It is not transparent to the user that the function may mask more points than specified.

Just some thoughts how to overcome this:
- *We could require that the remainint points are exactly the original points without the points to be masked.*

- *Maybe we should not allow in the interface to pass in points, but require a reduced mesh to be passed in. Such a mesh we could check to be valid and be a real masked mesh of the original reference. Maybe this could then guarantee that the outcome is exactly what is specified by the input.*

Personally, I like the second option more. Any other suggestion?

thogerig commented:
I like the second idea also the most. How about passing the MeshCompactifier instead of a PointId list?

```
  def mask(originalMomo: MoMo, reducedMesh: MeshCompactifier): MoMo = {

      val maskedMesh = reducedMesh.transformedMesh
      val remainingPtIds = maskedMesh.pointSet.pointIds.map(id => reducedMesh.pointBackMap(id)).toIndexedSeq
      val maskedModelShape = originalMomo.neutralModel.shape.marginal(remainingPtIds)
      val maskedModelColor = originalMomo.neutralModel.color.marginal(remainingPtIds)
      MoMo(maskedMesh, maskedModelShape, maskedModelColor)
...
```

This is, however also not completely clean because the mesh given to the MeshCompactifier must be the models reference mesh.

thogerig committed:
Removed marginal operations from MoMo class.

thogerig committed:
Mask MoMo with PointIds or masked reference mesh.

thogerig commented:
After some discussions, here a revised implementation:
* Since it does not behave as the original marginal function, I removed them from the MoMo class and moved them over to ModelHelpers.
* In addition to the `Seq[PointId]` also the masked mesh itself is supported.

What do you think of this change?

Andreas-Forster committed:
added some checks and feedback for the user in case of failure

Andreas-Forster commented:
Thank you for your update. I think it might be a good idea to make the things that can go wrong more transparent and provide the user with feedback what went wrong. So I encapsulated the return type in a `Try` and provided feedback in erroneous situations. What do you think about this?

Andreas-Forster committed:
fixed a bug and added some tests

Andreas-Forster commented:
I added some tests and fixed a bug I discovered while developing the tests.

thogerig commented:
Thanks for the additions and the bug fix. From my side the changes are fine.
